features/breed_list/src/test/java/com/example/catbreeds/breed_list/unitTests/BreedListViewModelTest.kt

```diff
@@ -0,0 +1,217 @@
+package com.example.catbreeds.breed_list.unitTests
+
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
+import com.example.catbreeds.domain.models.Breed
+import com.example.catbreeds.domain.repository.BreedRepository
+import com.example.catbreeds.breed_list.BreedListViewModel
+import com.example.catbreeds.test_core.MainDispatcherRule
+import com.example.catbreeds.test_core.mock.getBreeds
+import io.mockk.coJustRun
+import io.mockk.coVerify
+import io.mockk.every
+import io.mockk.impl.annotations.RelaxedMockK
+import io.mockk.junit4.MockKRule
+import io.mockk.spyk
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.flowOf
+import kotlinx.coroutines.test.advanceUntilIdle
+import kotlinx.coroutines.test.runTest
+import org.junit.Assert.*
+import org.junit.Rule
+import org.junit.Test
+
+@ExperimentalCoroutinesApi
+class BreedListViewModelTest {
+
+    @get:Rule
+    val instantTaskExecutorRule = InstantTaskExecutorRule()
+    @get:Rule
+    val mainDispatcherRule = MainDispatcherRule()
+    @get:Rule
+    val mockkRule = MockKRule(this)
+
+    @RelaxedMockK
+    private lateinit var breedRepository: BreedRepository
+
+    private val testBreeds = getBreeds()
+
+    private val vmUnderTest: BreedListViewModel by lazy {
+        spyk(
+            BreedListViewModel(breedRepository)
+        )
+    }
+
+    // Helper Methods
+    private fun setupRepository(breeds: List<Breed>, favoriteBreeds: List<Breed> = emptyList()) {
+        every { breedRepository.getBreeds() } returns flowOf(breeds)
+        every { breedRepository.getFavoriteBreeds() } returns flowOf(favoriteBreeds)
+        coJustRun { breedRepository.refreshBreeds() }
+    }
+
+    private fun setupToggleFavorites(breedId: String) {
+        coJustRun { breedRepository.addBreedToFavorites(breedId) }
+        coJustRun { breedRepository.removeBreedFromFavorites(breedId) }
+    }
+
+    private fun verifyRefreshBreedsCalled(times: Int = 1) {
+        coVerify(exactly = times) { breedRepository.refreshBreeds() }
+    }
+
+    private fun verifyAddBreedToFavorites(breedId: String, times: Int = 1) {
+        coVerify(exactly = times) { breedRepository.addBreedToFavorites(breedId) }
+    }
+
+    private fun verifyRemoveBreedFromFavorites(breedId: String, times: Int = 1) {
+        coVerify(exactly = times) { breedRepository.removeBreedFromFavorites(breedId) }
+    }
+
+    // Tests
+    @Test
+    fun `WHEN initialized with empty repository SHOULD have empty state`() = runTest {
+        // GIVEN
+        setupRepository(emptyList())
+
+        // WHEN
+        val vm = vmUnderTest
```

Comment:
The new val is not needed. We just need to call `vmUnderTest` to initialize it, because it's lazy. Then we can use aways `vmUnderTest` class member variable.

```suggestion
        vmUnderTest
```

Reply:
Good idea. Made the changes on 7697e0f